src/tests/test_update.py

```diff
@@ -102,4 +102,36 @@ def test_statistics(self, mock):
         self.assertEqual(results['num_recompares'], 1)
         self.assertEqual(results['num_stored_events'], 1)
         self.assertEqual(results['num_skipped_events_skipped'], 1)
-        self.assertEqual(results['num_applied_applied'], 1)
\ No newline at end of file
+        self.assertEqual(results['num_applied_applied'], 1)
+
+    def test_gob_event_action(self, mock_event):
+        # setup initial event and data
+        dummy_event = fixtures.get_event_fixure()
+
+        last_event_expected = 1
+        for action_expected in ['ADD', 'DELETE', 'CONFIRM', 'MODIFY']:
+            data = {'_last_event': last_event_expected}
+            dummy_event.action = action_expected
+
+            # setup done, run gob event
+            gob_event = _get_gob_event(dummy_event, data)
+
+            # assert action and last_event are as expected
+            self.assertEqual(action_expected, gob_event.name)
+            self.assertEqual(last_event_expected, gob_event.last_event)
+
+            # Increase last event for next test
+            last_event_expected += 1
+
+    def test_gob_event_invalid_action(self, mock_event):
+        # setup initial event and data
+        dummy_event = fixtures.get_event_fixure()
+
+        last_event_expected = 1
+        for action_expected in ['FOO', 'BAR']:
+            data = {'_last_event': last_event_expected}
+            dummy_event.action = action_expected
+
+            # Assert that Exception is thrown when events have invalid actions
+            self.assertRaises(GOBException, _get_gob_event, dummy_event, data)
+            last_event_expected += 1
```

Comment:
This test can be written a little simpler probably:
What do you think of:
    def test_gob_event_invalid_action(self, mock_event):
        # setup initial event and data
        dummy_event = fixtures.get_event_fixure()

        for invalid_action in ['FOO', 'BAR', None, 1]:
            dummy_event.action = invalid_action

            # Assert that Exception is thrown when events have invalid actions
            self.assertRaises(GOBException, _get_gob_event, dummy_event, {})